rotkehlchen/globaldb/updates/parsers.py

```diff
@@ -0,0 +1,334 @@
+import abc
+import logging
+import re
+from collections.abc import Callable
+from typing import TYPE_CHECKING, Generic, TypeVar
+
+from rotkehlchen.assets.types import AssetData, AssetType
+from rotkehlchen.errors.asset import UnknownAsset
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_evm_address
+from rotkehlchen.types import ChainID, ChecksumEvmAddress, EvmTokenKind, Timestamp
+
+from .types import ParsedAssetData, VersionRange
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+
+    from rotkehlchen.db.drivers.gevent import DBConnection
+
+T = TypeVar('T')
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class BaseParser(abc.ABC, Generic[T]):
```

Comment:
Specify it's a parser for assets.

```suggestion
class BaseAssetParser(abc.ABC, Generic[T]):
```